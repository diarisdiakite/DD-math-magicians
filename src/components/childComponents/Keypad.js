import React from 'react';
import PropTypes from 'prop-types';

function Keypad({ handleClick /* handleClear, handleCalculate  */ }) {
  return (

    <div className="keypad">
      <div className="row">
        <button
          type="button"
          name="AC"
          className="key"
          onClick={() => handleClick('AC')}
        >
          AC
        </button>
        <button
          type="button"
          name="+/-"
          className="key"
          onClick={() => handleClick('+/-')}
        >
          +/-
        </button>
        <button
          type="button"
          name="%"
          className="key"
          onClick={() => handleClick('%')}
        >
          %
        </button>
        <button
          type="button"
          name="/"
          className="key operator"
          onClick={() => handleClick('รท')}
        >
          /
        </button>
      </div>
      <div className="row">
        <button
          type="button"
          name="7"
          className="key"
          onClick={() => handleClick('7')}
        >
          7
        </button>
        <button
          type="button"
          name="8"
          className="key"
          onClick={() => handleClick('8')}
        >
          8
        </button>
        <button
          type="button"
          name="9"
          className="key"
          onClick={() => handleClick('9')}
        >
          9
        </button>
        <button
          type="button"
          name="x"
          className="key operator"
          onClick={() => handleClick('x')}
        >
          x
        </button>
      </div>

      <div className="row">
        <button
          type="button"
          name="4"
          className="key"
          onClick={() => handleClick('4')}
        >
          4
        </button>
        <button
          type="button"
          name="5"
          className="key"
          onClick={() => handleClick('5')}
        >
          5
        </button>
        <button
          type="button"
          name="6"
          className="key"
          onClick={() => handleClick('6')}
        >
          6
        </button>
        <button
          type="button"
          name="-"
          className="key operator"
          onClick={() => handleClick('-')}
        >
          -
        </button>
      </div>
      <div className="row">
        <button
          type="button"
          name="1"
          className="key"
          onClick={() => handleClick('1')}
        >
          1
        </button>
        <button
          type="button"
          name="2"
          className="key"
          onClick={() => handleClick('2')}
        >
          2
        </button>
        <button
          type="button"
          name="3"
          className="key"
          onClick={() => handleClick('3')}
        >
          3
        </button>
        <button
          type="button"
          name="+"
          className="key operator"
          onClick={() => handleClick('+')}
        >
          +
        </button>
      </div>
      <div className="row">
        <button
          type="button"
          name="0"
          className="key zero"
          onClick={() => handleClick('0')}
        >
          0
        </button>
        <button
          type="button"
          name="."
          className="key"
          onClick={() => handleClick('.')}
        >
          .
        </button>
        <button
          type="button"
          name="="
          className="key operator"
          onClick={handleClick}
        >
          =
        </button>
      </div>
    </div>
  );
}

Keypad.propTypes = {
  handleClick: PropTypes.func.isRequired,
};

export default Keypad;
