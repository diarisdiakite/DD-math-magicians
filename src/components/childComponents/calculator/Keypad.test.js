import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Keypad from './Keypad';

describe('Keypad component', () => {
  it('renders correctly', () => {
    const handleClickMock = jest.fn();

    const tree = renderer
      .create(
        <MemoryRouter>
          <Keypad handleClick={handleClickMock} />
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
