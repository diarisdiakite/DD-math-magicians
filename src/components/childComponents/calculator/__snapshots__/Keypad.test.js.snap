// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly 1`] = `
<div
  className="keypad"
>
  <div
    className="row"
  >
    <button
      className="key"
      name="AC"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="key"
      name="+/-"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="key"
      name="%"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="key operator"
      name="/"
      onClick={[Function]}
      type="button"
    >
      /
    </button>
  </div>
  <div
    className="row"
  >
    <button
      className="key"
      name="7"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="key"
      name="8"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="key"
      name="9"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="key operator"
      name="x"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
  </div>
  <div
    className="row"
  >
    <button
      className="key"
      name="4"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="key"
      name="5"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="key"
      name="6"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="key operator"
      name="-"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </div>
  <div
    className="row"
  >
    <button
      className="key"
      name="1"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="key"
      name="2"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="key"
      name="3"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="key operator"
      name="+"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>
  <div
    className="row"
  >
    <button
      className="key zero"
      name="0"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="key"
      name="."
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="key operator"
      name="="
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;

exports[`Keypad component renders correctly 1`] = `
<div
  className="keypad"
>
  <div
    className="row"
  >
    <button
      className="key"
      name="AC"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="key"
      name="+/-"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="key"
      name="%"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="key operator"
      name="/"
      onClick={[Function]}
      type="button"
    >
      /
    </button>
  </div>
  <div
    className="row"
  >
    <button
      className="key"
      name="7"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="key"
      name="8"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="key"
      name="9"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="key operator"
      name="x"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
  </div>
  <div
    className="row"
  >
    <button
      className="key"
      name="4"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="key"
      name="5"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="key"
      name="6"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="key operator"
      name="-"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </div>
  <div
    className="row"
  >
    <button
      className="key"
      name="1"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="key"
      name="2"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="key"
      name="3"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="key operator"
      name="+"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>
  <div
    className="row"
  >
    <button
      className="key zero"
      name="0"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="key"
      name="."
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="key operator"
      name="="
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
